### PR TITLE
feature: migrate all fetch endpoints to api/get endpoints

### PR DESCRIPTION
**Pull request by geclos**

### CODE REVIEW

File: apps/web/src/app/(private)/evaluations/(evaluation)/[evaluationUuid]/editor/import-logs/page.tsx [hunk 1]
```diff
@@ -147,7 +147,9 @@ const ProviderLogsTable = ({
               role='button'
             >
               <TableCell>
-                <Text.H4 noWrap>{relativeTime(log.generatedAt)}</Text.H4>
+                <Text.H4 noWrap>
+                  {relativeTime(new Date(log.generatedAt))}
+                </Text.H4>
```
[the remark this answers is not in the document]
Comment (csansoon) [reply apps/web/src/app/(private)/evaluations/(evaluation)/[evaluationUuid]/editor/import-logs/page.tsx:152]: Like [here](https://github.com/latitude-dev/latitude-llm/blob/fc5b98176c474ac6063fdae2d5530aa9c56ffe63/apps/web/src/stores/evaluationResultsByDocumentContent.ts#L68)

File: apps/web/src/app/(private)/datasets/_components/DatasetsTable/index.tsx [hunk 2]
```diff
@@ -30,6 +32,21 @@ export function DatasetsTable({
...
+        link={
+          <Link href={ROUTES.datasets.new.root}>
+            <TableBlankSlate.Button>
+              Create your first dataset
+            </TableBlankSlate.Button>
+          </Link>
+        }
+      />
+    )
+  }
```
Comment (geclos) [inline apps/web/src/app/(private)/datasets/_components/DatasetsTable/index.tsx:48]: doing some extra improvements of stuff i've found during this refactor

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/Metadata.tsx [hunk 3]
```diff
@@ -79,13 +79,6 @@ export function EvaluationResultMetadata({
 
   return (
     <>
-      <MetadataItem label='Result id'>
-        <ClickToCopy copyValue={evaluationResult.id.toString()}>
-          <Text.H5 align='right' color='foregroundMuted'>
-            {evaluationResult.id}
-          </Text.H5>
-        </ClickToCopy>
-      </MetadataItem>
```
Comment (geclos) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/Metadata.tsx:88]: useless so removing it

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx [hunk 4]
```diff
@@ -97,7 +97,7 @@ export function EvaluationResultInfo({
             {selectedTab === 'messages' && (
               <EvaluationResultMessages providerLog={providerLog} />
             )}
-            <div className='w-full bg-white flex justify-center'>
```
Comment (geclos) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx:100]: dark mode bug

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/BigNumberPanels/ModalValuePanel/index.tsx [hunk 5]
```diff
@@ -30,7 +31,11 @@ export default function ModalValuePanel({
       fallbackData: modal,
     },
   )
-  const onStatusChange = useCallback(() => refetch(), [refetch])
+  const onStatusChange = useDebouncedCallback(
+    useCallback(() => refetch(), [refetch]),
+    2000,
+    { trailing: true },
+  )
```
Comment (geclos) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/BigNumberPanels/ModalValuePanel/index.tsx:38]: absolutely bonkers this was recalculating on each new eval result coming in real time resulting in a self-ddos

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/BigNumberPanels/TotalsPanels/index.tsx [hunk 6]
```diff
@@ -30,7 +31,11 @@ export default function TotalsPanels({
       fallbackData: aggregation,
     },
   )
-  const onStatusChange = useCallback(() => refetch(), [refetch])
+  const onStatusChange = useDebouncedCallback(
+    useCallback(() => refetch(), [refetch]),
+    2000,
+    { trailing: true },
+  )
```
Comment (geclos) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/BigNumberPanels/TotalsPanels/index.tsx:38]: same

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/CostOverResults.tsx [hunk 7]
```diff
@@ -21,13 +23,20 @@ export function CostOverResultsChart({
...
       evaluation,
       documentUuid,
     })
-  const { commit } = useCurrentCommit()
-  const onStatusChange = useCallback(() => refetch(), [refetch])
+  const onStatusChange = useDebouncedCallback(
+    useCallback(() => refetch(), [refetch]),
+    2000,
+    { trailing: true },
+  )
```
Comment (geclos) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/CostOverResults.tsx:39]: same

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/ResultsOverTime.tsx [hunk 8]
```diff
@@ -23,11 +29,19 @@ export function ResultOverTimeChart({
...
+    commitUuid: commit.uuid,
     evaluation,
     documentUuid,
   })
-  const onStatusChange = useCallback(() => refetch(), [refetch])
+  const onStatusChange = useDebouncedCallback(
+    useCallback(() => refetch(), [refetch]),
+    2000,
+    { trailing: true },
+  )
```
Comment (geclos) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/ResultsOverTime.tsx:44]: you get the idea

File: apps/web/src/app/(private)/evaluations/(evaluation)/[evaluationUuid]/editor/import-logs/page.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (geclos) [reply apps/web/src/app/(private)/evaluations/(evaluation)/[evaluationUuid]/editor/import-logs/page.tsx:152]: yeah it has to be in the store

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/logs/_components/DocumentLogs/DocumentLogInfo/Metadata.tsx [hunk 9]
```diff
@@ -83,13 +83,6 @@ export function DocumentLogMetadata({
 
   return (
     <>
-      <MetadataItem label='Log uuid'>
-        <ClickToCopy copyValue={documentLog.uuid}>
-          <Text.H5 align='right' color='foregroundMuted'>
-            {documentLog.uuid.split('-')[0]}
-          </Text.H5>
-        </ClickToCopy>
-      </MetadataItem>
```
Comment (geclos) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/logs/_components/DocumentLogs/DocumentLogInfo/Metadata.tsx:92]: useless removing

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/BigNumberPanels/ModalValuePanel/index.tsx (hunk 5 above)
Comment (andresgutgon) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/BigNumberPanels/ModalValuePanel/index.tsx:38]: If we have realtime maybe these heavy computational stats should be left stale forever by configuring the SWR store no?

Comment (geclos) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/BigNumberPanels/ModalValuePanel/index.tsx:38]: yeah probably

File: apps/web/src/hooks/useFetcher.ts [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+      }
+
+      return fallback
+    }
+
+    return serializer
+      ? serializer(await response.json())
+      : await response.json()
+  }
+}
```
[the remark this answers is not in the document]
Comment (andresgutgon) [reply apps/web/src/hooks/useFetcher.ts:52]: credentials: include is not needed?

File: apps/web/src/stores/documentLogs.ts [hunk 11]
```diff
@@ -21,45 +19,12 @@ export default function useDocumentLogs(
...
-        title: 'Error',
-        description: (error as Error).message,
-        variant: 'destructive',
-      })
-
-      return []
-    }
-  }, [commitUuid, documentUuid, projectId, toast, page, pageSize])
+  const fetcher = useFetcher(
+    `/api/projects/${projectId}/commits/${commitUuid}/documents/${documentUuid}/documentLogs?page=${page ?? undefined}&pageSize=${pageSize ?? undefined}`,
```
Comment (andresgutgon) [inline apps/web/src/stores/documentLogs.ts:23]: `useFetcher` could have a `searchParams` argument that accept an object and convert to string

Comment (andresgutgon) [reply apps/web/src/stores/documentLogs.ts:23]: Also use ROUTES

Comment (geclos) [reply apps/web/src/stores/documentLogs.ts:23]: 💯

File: apps/web/src/hooks/useFetcher.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (geclos) [reply apps/web/src/hooks/useFetcher.ts:52]: hmm curious, requests are not failing without it, anyhow i will add it just for safeties

File: apps/web/src/hooks/useFetcher.ts [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+function buildRoute(route: string, searchParams?: ISearchParams) {
+  if (!searchParams) return route
+
+  const params = new URLSearchParams(searchParams)
+  return route.toString() + '?' + params.toString()
```
Comment (andresgutgon) [inline apps/web/src/hooks/useFetcher.ts:72]: Nice

**Review by andresgutgon — APPROVED** [review]

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/BigNumberPanels/ModalValuePanel/index.tsx (hunk 5 above)
Comment (andresgutgon) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/BigNumberPanels/ModalValuePanel/index.tsx:38]: Do this please. I think it makes a lot of sense

**Pull request merged**